browser/ui/webui/settings/default_brave_shields_handler.cc

```diff
@@ -206,6 +212,23 @@ void DefaultBraveShieldsHandler::GetCookieControlType(
                             base::Value(ControlTypeToString(setting)));
 }
 
+void DefaultBraveShieldsHandler::GetHideBlockAllCookieFlag(
+    const base::Value::List& args) {
+  CHECK_EQ(args.size(), 1U);
+  CHECK(profile_);
+  const ControlType setting = brave_shields::GetCookieControlType(
+      HostContentSettingsMapFactory::GetForProfile(profile_),
+      CookieSettingsFactory::GetForProfile(profile_).get(), GURL());
+
+  const bool block_all_cookies_feature_enabled = base::FeatureList::IsEnabled(
```

Comment:
nit: when `block_all_cookies_feature_enabled` is false, no need to calculate `setting` value, because response would be hide in anyway. This is a minor thing and I leave up to author the decision to change it or not.

Reply:
resolved as I didn't understand the task fully